src/pysplunkhec.py

```diff
@@ -0,0 +1,89 @@
+# the following allows JSON to be posted to Splunk HEC (HTTP Event Collector)
+
+import os
+import requests
+import json
+from src.kms import decrypt
+
+class PySplunkHEC:
+    def __init__(
+        self,
+        hec_token,
+        hec_host,
+        index="main",
+        host="null",
+        source="null",
+        sourcetype="json",
+        hec_port='443'
+        ):
+        self.index = index
+        self.host = host
+        self.source = source
+        self.sourcetype = sourcetype
+        self.hec_token = hec_token
+        self.uri = "https://" + hec_host + ":" + hec_port + "/services/collector"
```

Comment:
Could probably do `f"https://{hec_host}:{hec_port}/services/collector"`.

If these aren't going to be changed much, could favour constants over initialising them here.

Reply:
this has been changed as requested, I think having the option of passing the hec_port and hec_host via env vars is preferable, incase we need to change things on the splunk side don the line